### PR TITLE
cmake: add selinux support

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@cbodley could you take a look?

@branto1 as 'fsadm_manage_pid()' is only found in fedora and rhel/centos, so i think it's okay to do so? @smithfarm what do you think, does SuSE has this interface defined?

File: CMakeLists.txt [hunk 1]
```diff
@@ -344,6 +344,13 @@ include_directories(${Boost_INCLUDE_DIRS})
 
 find_package(Threads REQUIRED)
 
+option(WITH_SELINUX "build SELinux policy" OFF)
```
Comment (smithfarm) [inline CMakeLists.txt]: @tchaikov Does this mean `WITH_SELINUX` is off by default?

**Comment by smithfarm** [comment]
@tchaikov I've never heard of anyone using SELinux with SUSE.

File: CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [inline CMakeLists.txt]: @smithfarm yes, `option()` is `OFF` by default. and i put `OFF` here to make it explicit.

**Comment by tchaikov** [comment]
thanks for your confirmation, Nathan!

**Comment by branto1** [comment]
@tchaikov I wonder, where have you been hitting the issues with fsadm_manage_pid? The macro is not available with older versions of SELinux policy packeges (like the ones in RHEL 6) but those should be fairly ancient by nowadays standards. Also, I'm not exactly sure whether distro_redhat covers fedora (although I think it does).

btw: Ignoring that line on non-redhat platforms would probably result in denials on that platform.

File: selinux/ceph.te [hunk 2]
```diff
@@ -106,7 +106,9 @@ allow ceph_t self:process setpgid;
 allow ceph_t var_run_t:dir { write create add_name };
 allow ceph_t var_run_t:file { write create open getattr };
 
+ifdef(`distro_redhat', `)
```
Comment (branto1) [inline selinux/ceph.te]: There is a typo, here: the ')' character at the end of the line should not be there.

Comment (tchaikov) [inline selinux/ceph.te]: oh, thanks!

**Comment by tchaikov** [comment]
@branto1 

i was trying to compile my patch on a debian box, hence this change. i had compilation error if `fsadm_manage_pid` is not defined.

> btw: Ignoring that line on non-redhat platforms would probably result in denials on that platform.

i understand the risk. will drop this change anyway. maybe just build failure with '-DWITH_SELINUX=ON' on a not-supported distro would be better than failure at run-time.

**Comment by tchaikov** [comment]
changelog

- dropped the change to ceph.te.

File: CMakeLists.txt [hunk 3]
```diff
@@ -344,6 +344,13 @@ include_directories(${Boost_INCLUDE_DIRS})
...
+if(WITH_SELINUX)
+  find_file(SELINUX_MAKEFILE selinux/devel/include/Makefile
+    PATH /usr/share)
```
Comment (cbodley) [inline CMakeLists.txt]: should it be a fatal error if we don't find this?

**Comment by tchaikov** [comment]
changelog

- bail out with FATAL_ERROR if SELinux's Makefile is not found.

@cbodley thanks. fixed, tested and re-pushed

**Comment by tchaikov** [comment]
@cbodley ping

**Comment by tchaikov** [comment]
i just ported the selinux/Makefile.am to cmake and the ceph.pp file is generated as expected. also they artifacts are installed just fine.

**Pull request merged**